src/vault.cairo

```diff
@@ -0,0 +1,121 @@
+#[starknet::interface]
+trait IVaultContract<TContractState> {
+    fn store_liquidity(ref self: TContractState, amount: u256);
+    fn withdraw_liquidity(ref self: TContractState, amount: u256);
+}
+
+#[starknet::contract]
+mod Vault {
+    use starknet::storage::{
+        StoragePointerReadAccess, StoragePointerWriteAccess, StoragePathEntry, Map
+    };
+    use openzeppelin::token::erc20::interface::{IERC20Dispatcher, IERC20DispatcherTrait};
+
+        use starknet::ContractAddress;
+        use openzeppelin::access::ownable::OwnableComponent;
+    use openzeppelin::upgrades::UpgradeableComponent;
+    use openzeppelin::upgrades::interface::IUpgradeable;
+
+    use starknet::ClassHash;
+
+    component!(path: OwnableComponent, storage: ownable, event: OwnableEvent);
+    component!(path: UpgradeableComponent, storage: upgradeable, event: UpgradeableEvent);
+
+    /// Ownable
+    #[abi(embed_v0)]
+    impl OwnableImpl = OwnableComponent::OwnableImpl<ContractState>;
```

Comment:
We need to implement a two-step transfer functionality in case the owner needs to be changed. [Docs](https://docs.openzeppelin.com/contracts-cairo/0.19.0/access#two_step_transfer)